cpu/cpu_windows_amd64.go

```diff
@@ -209,5 +207,6 @@ func computeCoresAndProcessors() (cpuInfo CPU_INFO, err error) {
 		}
 	}
 
+	err = nil
```

Comment:
Why do we overwrite `err` here ? Isn't it already nil or contained an error we want to return ?

Reply:
The thing is that windows syscalls in go return a non-nil error in case of success (as well as a numerical "status" which has to be checked to know if there was actually an error)... 

In particular here we write the result of `getProcInfo.Call` directly into the `err` which is returned, so at the end of the function we would return this "success error" if we didn't overwrite it, which means that when checking whether the `computeCoresAndProcessors` succeeds one would think it did not.

Now that I look at it again I guess we could just not write the result of `getProcInfo.Call` in `err`, that would be clearer, what do you think @hush-hush ?

Reply:
I think it would be clearer indeed. Also, maybe not naming the return variable in the function signature might be better (and we would simply `return nil`).

But thanks for the explanation, make more sens !